adlfs/spec.py

```diff
@@ -1944,13 +1951,14 @@ async def _async_upload_chunk(self, final: bool = False, **kwargs):
         block_id = f"{block_id:07d}"
         if self.mode == "wb":
             try:
-                async with self.container_client.get_blob_client(blob=self.blob) as bc:
-                    await bc.stage_block(
-                        block_id=block_id,
-                        data=data,
-                        length=length,
-                    )
-                self._block_list.append(block_id)
+                for chunk in self.get_chunks(data):
+                    async with self.container_client.get_blob_client(blob=self.blob) as bc:
+                        await bc.stage_block(
+                            block_id=block_id, data=chunk, length=len(chunk),
+                        )
+                        self._block_list.append(block_id)
+                        block_id = len(self._block_list)
+                        block_id = f"{block_id:07d}"
```

Comment:
What does this change do?

Reply:
This is to update the ```block_id``` with a new value for the next chunk.